### PR TITLE
Add permissions needed for the "Mesh" page

### PR DESCRIPTION
**Pull request by israel-hdez**
Related kiali/kiali#1833
Related kiali/kiali#3409

### CODE REVIEW

**Comment by jmazzitelli** [comment]
I'm confused about this PR. I think it is doing it wrong unless I misunderstand why you need a completely separate role. FWIW, adding a new role/binding means we now have to create this role in addition to the existing core role in each namespace Kiali has access to - and this is going to make the operator much slower.

I think you can add this "list secrets" permission with the rest of permissions in the core role - follow the checklist here: [Are You Altering a Kiali Server Role Permission?](https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-role-permission)

**Comment by jmazzitelli** [comment]
My previous comment is based on the assumption that Kiali Server needs this list-secrets permission across all accessible namespaces. I am wondering if that is a bad assumption on my part.

If it is true that you only need this permission in the control plane namespace itself (e.g. istio-system) then my only request is you rename the role and role binding. I want this because this sounds like something we might need again in the future and it might have nothing to do with multi-cluster.

So can you just rename the role and role-binding (both the resources themselves and the yaml files) so they are more generic? I am thinking something like "role-controlplane" and "rolebinding-controlplane" (or something similar). The point is - we may need to add to these "istio-system-namespace-only" permissions in the future, and they may not have anything to do with multi-cluster. So make the name more generic.

**Comment by jmazzitelli** [comment]
No matter how this is implemented, we need the helm-charts to also be updated as well: https://github.com/kiali/helm-charts/tree/master/kiali-server/templates

Also, because the operator needs to the same permissions it wants to grant, you have to add these to the operator helm-chart as well as the OLM metadata files.

Again, see the checklist for the different places to do this.

File: roles/default/kiali-deploy/tasks/kubernetes/k8s-main.yml [hunk 1]
```diff
@@ -7,7 +7,9 @@
   - serviceaccount
   - configmap
   - "{{ 'role-viewer' if kiali_vars.deployment.view_only_mode|bool == True else 'role' }}"
+  - role-multi-cluster
   - rolebinding
+  - rolebinding-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/tasks/kubernetes/k8s-main.yml:12]:
```suggestion
  - role-controlplane
  - rolebinding
  - rolebinding-controlplane
```

File: roles/default/kiali-deploy/tasks/openshift/os-main.yml [hunk 2]
```diff
@@ -8,7 +8,9 @@
   - configmap
   - cabundle
   - "{{ 'role-viewer' if kiali_vars.deployment.view_only_mode|bool == True else 'role' }}"
+  - role-multi-cluster
   - rolebinding
+  - rolebinding-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/tasks/openshift/os-main.yml:13]:
```suggestion
  - role-controlplane
  - rolebinding
  - rolebinding-controlplane
```

File: roles/default/kiali-deploy/templates/kubernetes/role-multi-cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kiali-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/kubernetes/role-multi-cluster.yaml:4]:
```suggestion
  name: kiali-controlplane
```

File: roles/default/kiali-deploy/templates/kubernetes/rolebinding-multi-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kiali-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/kubernetes/rolebinding-multi-cluster.yaml:4]:
```suggestion
  name: kiali-controlplane
```

File: roles/default/kiali-deploy/templates/kubernetes/rolebinding-multi-cluster.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  labels:
+    app: kiali
+    version: {{ kiali_vars.deployment.version_label }}
+    app.kubernetes.io/name: kiali
+    app.kubernetes.io/version: {{ kiali_vars.deployment.version_label }}
+    app.kubernetes.io/part-of: kiali
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kiali-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/kubernetes/rolebinding-multi-cluster.yaml:15]:
```suggestion
  name: kiali-controlplane
```

Comment (jmazzitelli) [reply roles/default/kiali-deploy/templates/kubernetes/rolebinding-multi-cluster.yaml:15]: rename this file `rolebinding-controlplane.yaml`

File: roles/default/kiali-deploy/templates/openshift/role-multi-cluster.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kiali-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/openshift/role-multi-cluster.yaml:4]:
```suggestion
  name: kiali-controlplane
```

File: roles/default/kiali-deploy/templates/openshift/rolebinding-multi-cluster.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kiali-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/openshift/rolebinding-multi-cluster.yaml:4]:
```suggestion
  name: kiali-controlplane
```

File: roles/default/kiali-deploy/templates/openshift/rolebinding-multi-cluster.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  labels:
+    app: kiali
+    version: {{ kiali_vars.deployment.version_label }}
+    app.kubernetes.io/name: kiali
+    app.kubernetes.io/version: {{ kiali_vars.deployment.version_label }}
+    app.kubernetes.io/part-of: kiali
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kiali-multi-cluster
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/templates/openshift/rolebinding-multi-cluster.yaml:15]:
```suggestion
  name: kiali-controlplane
```

Comment (jmazzitelli) [reply roles/default/kiali-deploy/templates/openshift/rolebinding-multi-cluster.yaml:15]: rename this file `rolebinding-controlplane.yaml`

File: roles/default/kiali-remove/tasks/remove-clusterroles.yml [hunk 9]
```diff
@@ -18,7 +18,9 @@
   - k8s_item.metadata.name is defined
   with_items:
   - "{{ query('k8s', kind='ClusterRoleBinding', resource_name='kiali', api_version='rbac.authorization.k8s.io/v1') }}"
+  - "{{ query('k8s', kind='RoleBinding', resource_name='kiali-multi-cluster', api_version='rbac.authorization.k8s.io/v1') }}"
   - "{{ query('k8s', kind='ClusterRole', resource_name='kiali', api_version='rbac.authorization.k8s.io/v1') }}"
   - "{{ query('k8s', kind='ClusterRole', resource_name='kiali-viewer', api_version='rbac.authorization.k8s.io/v1') }}"
+  - "{{ query('k8s', kind='Role', resource_name='kiali-multi-cluster', api_version='rbac.authorization.k8s.io/v1') }}"
```
Comment (jmazzitelli) [inline roles/default/kiali-remove/tasks/remove-clusterroles.yml:24]:
```suggestion
  - "{{ query('k8s', kind='RoleBinding', resource_name='kiali-controlplane', api_version='rbac.authorization.k8s.io/v1') }}"
  - "{{ query('k8s', kind='ClusterRole', resource_name='kiali', api_version='rbac.authorization.k8s.io/v1') }}"
  - "{{ query('k8s', kind='ClusterRole', resource_name='kiali-viewer', api_version='rbac.authorization.k8s.io/v1') }}"
  - "{{ query('k8s', kind='Role', resource_name='kiali-controlplane', api_version='rbac.authorization.k8s.io/v1') }}"
```

File: roles/default/kiali-deploy/templates/kubernetes/role-multi-cluster.yaml (hunk 3 above)
Comment (jmazzitelli) [reply roles/default/kiali-deploy/templates/kubernetes/role-multi-cluster.yaml:4]: rename this file `role-controlplane.yaml`

File: roles/default/kiali-deploy/templates/openshift/role-multi-cluster.yaml (hunk 6 above)
Comment (jmazzitelli) [reply roles/default/kiali-deploy/templates/openshift/role-multi-cluster.yaml:4]: rename this file `role-controlplane.yaml`

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Comment by israel-hdez** [comment]
@jmazzitelli I think this PR is ready. I'm going to prepare a PR against the helm-charts repo.

**Review by jmazzitelli — APPROVED** [review]
lgtm

**Pull request merged**